lib/resty/wasmx/shm.lua

```diff
@@ -355,12 +366,37 @@ local function metrics_define(zone, name, metric_type)
         error(err, 2)
     end
 
+    local cbins = nil
+    local n_bins = 0
+
+    if opts ~= nil then
+        if type(opts) ~= "table" then
+            error("opts must be a table", 2)
+        end
+
+        if opts.bins ~= nil then
+            if type(opts.bins) ~= "table" then
+                error("opts.bins must be a table", 2)
+            end
+
+            if #opts.bins > HISTOGRAM_MAX_BINS then
+                local err = "opts.bins must have up to %d numbers"
+                error(str_fmt(err, HISTOGRAM_MAX_BINS), 2)
+            end
+
+            sort(opts.bins)
```

Comment:
There is no guarantee up to this point that the `opts.bins` elements are numbers, or that they are positive either. We should probably add some checks?

Reply:
Good point. I'll add validation for the contents of `opts.bins` as well.

Reply:
Can we remove the call to `sort` now?

Reply:
Oh, my bad.